lms/djangoapps/ccx/api/v0/tests/test_views.py

```diff
@@ -536,6 +580,53 @@ def test_authorization(self):
         resp = self.client.get(self.detail_url, {}, HTTP_AUTHORIZATION=self.auth)
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
 
+    def test_authorization_no_oauth(self):
+        """
+        Check authorization for users logged in without oauth
+        """
+        # create a staff user
+        staff_user = User.objects.create_user('test_staff_user', 'test_staff_user@openedx.org', 'test')
+        # add staff role to the app user
+        CourseStaffRole(self.master_course_key).add_users(staff_user)
+        # create an instructor user
+        instructor_user = User.objects.create_user('test_instructor_user', 'test_instructor_user@openedx.org', 'test')
+        # add staff role to the app user
+        CourseInstructorRole(self.master_course_key).add_users(instructor_user)
+        # create an coach user
+        coach_user = User.objects.create_user('test_coach_user', 'test_coach_user@openedx.org', 'test')
+        # add staff role to the app user
+        CourseCcxCoachRole(self.master_course_key).add_users(instructor_user)
+
+        data = {'display_name': 'CCX Title'}
+        # the staff user can perform the request
+        self.client.login(username=staff_user.username, password='test')
+        resp = self.client.get(self.detail_url)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        resp = self.client.patch(self.detail_url, data, format='json')
+        self.assertEqual(resp.status_code, status.HTTP_204_NO_CONTENT)
+        self.client.logout()
+        # the instructor user can perform the request
```

Comment:
Do we need to do anything to reset the state in between logout and login, to avoid one code block influencing the later code block?